### PR TITLE
Add 'System' flag to User model

### PR DESCRIPTION
**Pull request by sirenChen**
Please see tickets 4119 in Product Backlog.
Add 'System' flag to User model.

### CODE REVIEW

File: src/foam/nanos/auth/User.js [hunk 1]
```diff
@@ -286,6 +286,11 @@ foam.CLASS({
       class: 'DateTime',
       name: 'lastModified',
       documentation: 'Last modified date.'
+    },
+    {
+      class: 'Boolean',
+      name: 'system',
+      documentation: 'Indicate system accounts.'
```
Comment (jlhughes) [inline src/foam/nanos/auth/User.js:294]: please explicitly default to false:
value: false

**Comment by jlhughes** [comment]
If possible, please create a TestCase.  The foam side testing is a little different, you can inquire on the foam channel if you can't find examples.

File: src/foam/nanos/auth/User.js [hunk 2]
```diff
@@ -290,6 +290,7 @@ foam.CLASS({
     {
       class: 'Boolean',
       name: 'system',
+      value: false,
```
Comment (kgrgreer) [inline src/foam/nanos/auth/User.js:293]: We already have the 'type' property. Couldn't we just add 'system' as a new type?

File: src/foam/nanos/auth/AuthenticatedUserDaoTest.java [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+
+
+    //test case
+    AuthenticatedUserDAO_AdminUpdateSystemFlag(adminContext);
+    AuthenticatedUserDAO_AdminCreateUserWithSystemFlag(adminContext);
+    AuthenticatedUserDAO_NonAdminUpdateSystemFlag(nonAdminContext);
+    AuthenticatedUserDAO_NonAdminCreateUserWithSystemFlag(nonAdminContext);
+  }
+
+  private void AuthenticatedUserDAO_AdminUpdateSystemFlag(X adminContext) {
```
Comment (kgrgreer) [inline src/foam/nanos/auth/AuthenticatedUserDaoTest.java]: Don't make methods private. Just makes harder to test.

File: src/foam/nanos/auth/AuthenticatedUserDaoTest.java [hunk 4]
```diff
@@ -0,0 +1,145 @@
+package foam.nanos.auth;
+
+import foam.core.X;
+import foam.dao.DAO;
+import foam.dao.MDAO;
+import foam.test.TestUtils;
+import foam.util.Auth;
+import foam.util.SafetyUtil;
+
+public class AuthenticatedUserDaoTest extends foam.nanos.test.Test {
```
Comment (kgrgreer) [inline src/foam/nanos/auth/AuthenticatedUserDaoTest.java]: Dao -> DAO

File: src/foam/nanos/auth/User.js (hunk 2 above)
Comment (jlhughes) [reply src/foam/nanos/auth/User.js:293]: @kgrgreer  We plan to sub-class User into these types such as BusinessUser so the type as it exists now will go away.

**Review by jlhughes — COMMENTED** [review]
comment deleted

File: src/foam/nanos/auth/User.js (hunk 1 above)
Comment (jlhughes) [reply src/foam/nanos/auth/User.js:294]: defaults to false, no need.

**Pull request merged**